cpp/src/arrow/compute/key_map_internal.cc

```diff
@@ -281,7 +281,7 @@ void SwissTable::early_filter_imp(const int num_keys, const uint32_t* hashes,
 // When we reach this limit, we need to break processing of any further rows and resize.
 //
 uint64_t SwissTable::num_groups_for_resize() const {
-  // Resize small hash tables when 50% full (up to 12KB).
+  // Resize small hash tables when 50% full (up to 32KB).
```

Comment:
Sorry, how to compute `32KB`?

Reply:
The `log_blocks_small = 9` means that we consider `1 << 9 = 512` blocks as "small". A block is a swiss-table-specific concept explained in https://github.com/apache/arrow/blob/main/cpp/src/arrow/acero/doc/key_map.md , which is 64-byte. So the small hash table ends up with `512 * 64B = 32KB`.

So the original `12KB` is obviously a typo.

Reply:
Thanks!